docs/NooBaaNonContainerized/S3Ops.md

```diff
@@ -40,6 +41,55 @@ The following lists describe the bucket and object operations available in NooBa
 - S3 PutObjectTagging
 - S3 DeleteObjectTagging
 
+### Bucket Policy Support
+Bucket policy can be added to a bucket using the S3 API or the noobaa-cli.
+
+**Prerequisites**  
+An account and an bucket, for bucket policy management using the S3 API you would use the NooBaa service. 
+
+#### Bucket Policy in NooBaa CLI
+1. Adding a bucket policy:
+  - On bucket creation using the command: `noobaa-cli bucket add --name <bucket_name> --owner <owner_name> --path <path> --bucket_policy <bucket-policy>`.
+  - On bucket update using the command: `noobaa-cli bucket update --name <bucket_name> --bucket_policy <bucket-policy>`.
+In both cases the argument for the bucket policy is a string
+2. Removing a bucket policy: `noobaa-cli bucket update --name <bucket_name> --bucket_policy ''` (using empty string)
+3. Get a bucket policy: `noobaa-cli bucket status --name <bucket_name>` (a bucket policy will be printed in the bucket details)
+
+bucket_policy as a string example:
+`'{"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"AWS":["<account-name>"]},"Action":["s3:*"],"Resource":["arn:aws:s3:::<bucket-name>/*","arn:aws:s3:::<bucket-name>"]}]}'`
+
+Replace `<account-name>` with account name and `<bucket-name>` with a bucket name
+
+#### Bucket Policy in S3 API (using AWS CLI)
+1. Adding bucket policy: `AWS_ACCESS_KEY_ID={access_key} AWS_SECRET_ACCESS_KEY={secret_key} aws s3api put-bucket-policy --endpoint-url {endpoint_address} --bucket {bucket_name} --policy file://policy.json`
+2. Removing a bucket policy: `AWS_ACCESS_KEY_ID={access_key} AWS_SECRET_ACCESS_KEY={secret_key} aws s3api delete-bucket-policy --endpoint-url {endpoint_address} --bucket {bucket_name} --policy file://policy.json`
+3. Get a bucket policy: `AWS_ACCESS_KEY_ID={access_key} AWS_SECRET_ACCESS_KEY={secret_key} aws s3api delete-bucket-policy --endpoint-url {endpoint_address} --bucket {bucket_name} --policy file://policy.json`
+
+policy.json example:
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [ 
+    { 
+     "Effect": "Allow", 
+     "Principal": { "AWS": [ "<account-name>" ] }, 
+     "Action": [ "s3:*" ], 
+     "Resource": [ "arn:aws:s3:::<bucket-name>/*", "arn:aws:s3:::<bucket-name>" ] 
+    }
+  ]
+}
+```
+Replace `<account-name>` with account name and `<bucket-name>` with a bucket name
+
+##### Principal Field:
+Currently we support a couple of options:
+1. All principals: `"Principal": { "AWS": [ "*" ] }`
```

Comment:
is seems like everywhere and in all the examples Principal much be AWS: [ something], we also support it without AWS:
https://github.com/noobaa/noobaa-core/blob/aba0ab77437d5e05d0fa231ab4f8d4dbf7e29c8d/src/endpoint/s3/s3_bucket_policy_utils.js#L168